### PR TITLE
First pass at applying async for sub-resources

### PR DESCRIPTION
**Pull request by alexhung**
Using [errgroup](https://pkg.go.dev/golang.org/x/sync/errgroup) to manage error returning. Open to use other packages/patterns since I'm new to GoRoutine.

### CODE REVIEW

File: pkg/projects/resource_project_membership.go [hunk 1]
```diff
@@ -196,26 +197,16 @@ var updateMember = func(projectKey string, membershipType string, member Member,
...
-		err := deleteMember(projectKey, membershipType, member, m)
-		if err != nil {
-			errs = append(errs, err)
-		}
-	}
+		projectKey, membershipType, member, m := projectKey, membershipType, member, m
 
-	if len(errs) > 0 {
-		return fmt.Errorf("failed to delete members from project: %v", errs)
+		g.Go(func() error {
```
Comment (chb0github) [inline pkg/projects/resource_project_membership.go:206]: I wish this library took a func param of `func() (interface{}, error)` that way your end result would be 2 slices:

```go
[]interface{}, []error
```
corresponding to each call

**Review by chb0github — APPROVED** [review]
Does it actually improve performance?

**Comment by alexhung** [comment]
@chb0github Nothing noticeable with the test, I suspect the setup/teardown of repos/users/groups mask any performance gain 😄

File: pkg/projects/resource_project_membership.go (hunk 1 above)
Comment (alexhung) [reply pkg/projects/resource_project_membership.go:206]: I agree. More like `Promise.allSettled()` in JS.

**Pull request merged**